src/components/app/data/hooks/useBrowseAndRequest.test.js

```diff
@@ -0,0 +1,165 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { QueryClientProvider } from '@tanstack/react-query';
+import { AppContext } from '@edx/frontend-platform/react';
+import { authenticatedUserFactory, enterpriseCustomerFactory } from '../services/data/__factories__';
+import useEnterpriseCustomer from './useEnterpriseCustomer';
+import { queryClient } from '../../../../utils/tests';
+import { fetchBrowseAndRequestConfiguration, fetchCouponCodeRequests, fetchLicenseRequests } from '../services';
+import useBrowseAndRequest, {
+  useBrowseAndRequestConfiguration,
+  useCouponCodeRequests,
+  useSubscriptionLicenseRequests,
+} from './useBrowseAndRequest';
+
+jest.mock('./useEnterpriseCustomer');
+jest.mock('../services', () => ({
+  ...jest.requireActual('../services'),
+  fetchBrowseAndRequestConfiguration: jest.fn().mockResolvedValue(null),
+  fetchLicenseRequests: jest.fn().mockResolvedValue(null),
+  fetchCouponCodeRequests: jest.fn().mockResolvedValue(null),
+}));
+
+const mockEnterpriseCustomer = enterpriseCustomerFactory();
+const mockAuthenticatedUser = authenticatedUserFactory();
+const mockBrowseAndRequestConfiguration = {
+  id: 123,
+};
+const mockLicenseRequests = ['test-license1', 'test-license2'];
+const mockCouponCodeRequests = ['test-couponCode1, test-couponCode2'];
+
+const Wrapper = ({ children }) => (
+  // eslint-disable-next-line react/jsx-filename-extension
+  <QueryClientProvider client={queryClient()}>
+    <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
+      {children}
+    </AppContext.Provider>
+
+  </QueryClientProvider>
+);
+
+describe('useBrowseAndRequestConfiguration', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    useEnterpriseCustomer.mockReturnValue({ data: mockEnterpriseCustomer });
+  });
+  it('should handle resolved value correctly', async () => {
+    fetchBrowseAndRequestConfiguration.mockResolvedValue(mockBrowseAndRequestConfiguration);
+    const { result, waitForNextUpdate } = renderHook(() => useBrowseAndRequestConfiguration(), { wrapper: Wrapper });
+    await waitForNextUpdate();
+
+    expect(result.current).toEqual(
+      expect.objectContaining({
+        data: mockBrowseAndRequestConfiguration,
+        isLoading: false,
+        isFetching: false,
+      }),
+    );
+  });
+  it('should handle rejected value correctly', async () => {
```

Comment:
[suggestion] I believe this test may be largely testing the functionality of `useQuery` more so than our implementation/usage of it. Relayed, `fetchBrowseAndRequestConfiguration` itself has a try/catch so I'm not sure it's intended this hook should ever be in an error state, unless an exception is thrown outside of the try/catch.

I might recommend removing this test case (and others similar to it below)? Curious to hear what you think.

Reply:
I had started writing out tests with an error state checking specifically around the use of `useQuery's` return of `isError`. But as I continued writing subsequent tests, I noticed the need for it became redundant and didn't include those tests in a few of the tests files. 
Your comment backs up the doubt I had in writing those tests further down the road, specifically because we are avoiding the use of the `isError` state by ensuring the data's existence before page load.

I am opting to remove those test cases that return an error state based on your feedback and not all the tests added in this PR end up checking for error state. 👍🏽